### PR TITLE
Fix miscellaneous bugs

### PR DESCRIPTION
**Pull request by pedrofigro**
- QScore (rebasing is pending)
   - [x] `run_experiment` does not work for a real backend because of the output folder name format: `"OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect"`
   - [x] Suppress (most) outputs and only show the plot
- GHZ
   - [x] change `experiment_all` notebook code to show the results as in `ghz_example`
- IRB
   - [x] Fix typo in avg fidelity plot legend

### CODE REVIEW

File: src/iqm/benchmarks/benchmark_experiment.py [hunk 1]
```diff
@@ -52,15 +53,14 @@ def __init__(
...
-
         if isinstance(backend, str):
             self.backend = get_iqm_backend(backend)
-
         else:
             assert isinstance(backend, IQMBackendBase)
             self.backend = backend
 
+        self.device_id = device_id if device_id is not None else self.backend.name
+
```
Comment (rabrie) [inline src/iqm/benchmarks/benchmark_experiment.py:63]: Is this what fixes the output folder name error?

File: examples/example_experiment_all.ipynb [hunk 2]
```diff
@@ -675,19 +673,88 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "for qubits, data in result_ghz.observations.items():\n",
-    "    print(f\"Qubits {qubits}:\")\n",
-    "    for k,v in data.items():\n",
-    "        print(f\"\\t{k}: {v}\")"
+    "# Show observations for a specific qubit layout\n",
+    "qubit_layout = [0,1,2]\n",
```
Comment (rabrie) [inline examples/example_experiment_all.ipynb:677]: FYI: I changed only the qubit layout in this cell

**Review by rabrie — APPROVED** [review]
Looking good!

File: src/iqm/benchmarks/benchmark_experiment.py (hunk 1 above)
Comment (pedrofigro) [reply src/iqm/benchmarks/benchmark_experiment.py:63]: Yes indeed, before it was trying to assign `backend` when it was passed as a backend object (and below the folder name has `device_id` in it)

File: examples/example_experiment_all.ipynb (hunk 2 above)
Comment (pedrofigro) [reply examples/example_experiment_all.ipynb:677]: Thanks! OK yes I wasn't sure what exactly was the update needed

**Pull request merged**